### PR TITLE
Remove flaky tests that keep failing randomly

### PR DESCRIPTION
**Pull request by 0nko**
This PR removes flaky tests that often fail randomly without a reason.

**To test:**
All green lights are enough.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by pachlava** [comment]
Hey @0nko 👋 🙂 

There recently were two influxes of fails:

1. I missed prolonging the plan for the store that is used in real API tests, and the store went down for about a day. All real API UI tests were failing during this period, unfortunately. This was totally my guilt. It was addressed by #11209.
2. The other was due to decreased Emulator performance on Firebase, which caused multiple fails of `e2eRealApiProductsSearchUsual `. The discussion is here - p1712695115551579-slack-C5ALGJYEP -, and it was solved with #11296.

Some older branches still have `e2eRealApiProductsSearchUsual` failing, because they do not contain the fix for `2`. Could you please let me know if you mean these cases, or there are other ones? Thank you!

**Review by JorgeMucientes — APPROVED** [review]

**Comment by JorgeMucientes** [comment]
Hey @pachlava thanks for sharing the context 👍🏼 
From a quick look at the PRs you linked, it seems both were merged more than 1 week ago. However, the instrumented tests keep failing consistently, even in PRs that have been updated with the `trunk` code today. For example https://github.com/woocommerce/woocommerce-android/pull/11318 -> [Instrumented tests failing](https://buildkite.com/automattic/woocommerce-android/builds/20294#018eedc3-d9cd-4793-8251-fb731fb5f821)

Maybe there are still some unresolved issues??

**Comment by pachlava** [comment]
> For example https://github.com/woocommerce/woocommerce-android/pull/11318 -> [Instrumented tests failing](https://buildkite.com/automattic/woocommerce-android/builds/20294#018eedc3-d9cd-4793-8251-fb731fb5f821)

Thank you for the link @JorgeMucientes 👋 In this particular case, this looks like an actual crash or freeze of the app. From the [video](https://console.firebase.google.com/u/1/project/api-project-108380595987/testlab/histories/bh.edfd947f2636efe3/matrices/6072354433020376470/executions/bs.1b9d24da4e58286d/testcases/41/video), it's seen that the test did not even get to the area of the test (Orders screen). [This log file](https://ff7fe453a548ca91995aec9b52eecc10b784e06dab269cc5c771326-apidata.googleusercontent.com/download/storage/v1/b/test-lab-wjdmcn8vd90jx-wfb9uburfx80m/o/2024-04-17_20:41:31.926379_wzPo%2FPixel2.arm-30-en-portrait%2Ftest_cases%2F0040_logcat?jk=AWNiL9IrdR_3UH5SnA69_zI9HdrDrPwgZR5gtk7INyFbfvr4icQUP8OjMT7RNYHvJ6XMcZw1lBSod4IQ8lIi3hV0MPc_80L7K8wmBlSeTxUqvZN-Avo1Ku0_b9n3P8jEoUaCNYsf8nONbhq-0_drqS2JwtzjTmD_z-2MBjk63NIoYvzl9Jr3aQtYMrQU9DZfIWQe_htBevIZUTFrAxT4LzHRThQUWnhlAU82qYsTr_lZH3af5ESUwt_5AVAASb3Wwm4Fl0tbEdMs1YZePG-KbWqFIb7LH-1byJMiD0JwCzjiG_ObcstkxfEZRf88OFdUcrJVyaqKk-KH7s2WBq4RrPaLgbRnXjsXclyYRDwpfdiin766_5vIwUSNGUCLWtghohw7-dTRHnm4S2L-8BpnKyMzECkEr2KfQyk6LfdnUgIw_mtk2s2sMHqMxUq7AZsa7TFonx2uJJ9pp7oOipeLnS3TFeNbjYMTOWZEnRw2gEBs3UnsYNTYMapJMi5D3I1dBdBLiN5otOOSMKcRMQAMNQvGDI01QkvkUZFaHt4I5NANYAnbbQxX2uqDBGEUsr-GfZZFVWR-uDvEfcBEu66rGS-axepdf2Eysntr6kBOqUHF0LgoAGTf36STTRHzNnrSN-MS8GnuWn2U8b4KH_5Xq5GxjkUmpqNb5TIPkol7w9s1oNY2uMcINL4WV-uBKiURBUDqQm8fwvkqNqBcARdjB8l038QuLzjOdYV0Ue3ymHJj7Dk-rs9AeUSeXgFC0eNbpek256-tm1yLt5CnlZsQwdn5VDggDbWa19zVL-92rymU3amp8nQSw658xDrhunkqtjsarMdWPTVjQKxm4GnKjp1LEcJyaP6UeIBbHFXiu7NeqsuIS8S01qTQCIfeSFDVBGSMD7YodcBdiZN0ZGGO98vNopFOWfJm604OQvTH6NomPyblT_7u41A9BrDcS3pfu_pJjcfQpiq9N6uezY0ZyyLUM79m44P5kH0HnZkFxMvu_ljh8YHqt7FXk1nsaSwBp-Ygm10w5j8FJpBeMrDcjRZInwF9Vwz50W95lM4lMrJtvH_-3haz2AheDhdjnO4azZEHKPZf950lkkyRcPgTjfNccr_eXTs0LTgghko2fq1TnMw3db0bw9fsYQuvewXRYnRzp8VGewp5hiBhxOs0dqHY-JXIbQC3yFMnH-nnoce5JjyeBnp8Rk-ekwklruRq0t07Ebr3bVVs4LWf5LJE6Hg1a6gD2Az2O6KexTa8mo2N9EXz6HDeYejeiE2fKu9D043kyQ&isca=1) maybe gives more info 🤔. Might this be an actual issue?

```
04-17 13:44:01.333: D/Sentry(9007): Class not available:io.sentry.compose.gestures.ComposeGestureTargetLocator
04-17 13:44:01.333: D/Sentry(9007): java.lang.ClassNotFoundException: io.sentry.compose.gestures.ComposeGestureTargetLocator
04-17 13:44:01.333: D/Sentry(9007): 	at java.lang.Class.classForName(Native Method)
04-17 13:44:01.333: D/Sentry(9007): 	at java.lang.Class.forName(Class.java:454)
04-17 13:44:01.333: D/Sentry(9007): 	at java.lang.Class.forName(Class.java:379)
04-17 13:44:01.333: D/Sentry(9007): 	at io.sentry.android.core.LoadClass.loadClass(LoadClass.java:21)
04-17 13:44:01.333: D/Sentry(9007): 	at io.sentry.android.core.LoadClass.isClassAvailable(LoadClass.java:39)
04-17 13:44:01.333: D/Sentry(9007): 	at io.sentry.android.core.LoadClass.isClassAvailable(LoadClass.java:44)
04-17 13:44:01.333: D/Sentry(9007): 	at io.sentry.android.core.AndroidOptionsInitializer.initializeIntegrationsAndProcessors(AndroidOptionsInitializer.java:189)
04-17 13:44:01.333: D/Sentry(9007): 	at io.sentry.android.core.SentryAndroid.lambda$init$1(SentryAndroid.java:140)
04-17 13:44:01.333: D/Sentry(9007): 	at io.sentry.android.core.SentryAndroid$$ExternalSyntheticLambda0.configure(Unknown Source:8)
04-17 13:44:01.333: D/Sentry(9007): 	at io.sentry.Sentry.applyOptionsConfiguration(Sentry.java:196)
04-17 13:44:01.333: D/Sentry(9007): 	at io.sentry.Sentry.init(Sentry.java:166)
04-17 13:44:01.333: D/Sentry(9007): 	at io.sentry.android.core.SentryAndroid.init(SentryAndroid.java:87)
04-17 13:44:01.333: D/Sentry(9007): 	at io.sentry.android.core.SentryAndroid.init(SentryAndroid.java:70)
04-17 13:44:01.333: D/Sentry(9007): 	at com.automattic.android.tracks.crashlogging.internal.SentryErrorTrackerWrapper.initialize(SentryErrorTrackerWrapper.kt:14)
04-17 13:44:01.333: D/Sentry(9007): 	at com.automattic.android.tracks.crashlogging.internal.SentryCrashLogging.<init>(SentryCrashLogging.kt:36)
04-17 13:44:01.333: D/Sentry(9007): 	at com.automattic.android.tracks.crashlogging.CrashLoggingProvider.createInstance(CrashLoggingProvider.kt:13)
04-17 13:44:01.333: D/Sentry(9007): 	at com.woocommerce.android.util.crashlogging.CrashLoggingModule$Companion.provideCrashLogging(CrashLoggingModule.kt:36)
```

UPD: this test (`e2eRealApiOrdersFilter`) consistently fails in the PR (#11318) branch, even locally. However, the same test doesn't fail on `trunk`.

**Comment by 0nko** [comment]
Sorry @pachlava, I was AFK last week.

Here's an example of a failed build from the last 2 weeks:

https://buildkite.com/automattic/woocommerce-android/builds/19967

There were a lot more (and recent) but I can't find them, it seems like the failed runs were replaced when I re-ran them.

**Comment by JorgeMucientes** [comment]
Sorry, I missed sharing some extra context after some discussion with @pachlava about the failing tests. 

TLDR: In my case, some instrumented tests were failing when I enabled the Feature flag `DYNAMIC_DASHBOARD` for debug builds because of the new compose UI. We will have to fix that when the feature is fully ready. As soon as I deactivated the feature flag, the tests passed. Leaving the full conversation screenshot below. 

However, the linked build from @0nko's latest https://github.com/woocommerce/woocommerce-android/pull/11327#issuecomment-2069250542 seems to be a different thing, where real flakiness comes into place. Maybe you might want to review those @pachlava ? 🙏🏼 

Context on prior conversation about test failing due to the new Compose UI

Screenshot 2024-04-22 at 17 56 27

**Comment by pachlava** [comment]
Thanks for providing more context @0nko 👋 

From having a look into the [recent runs on Firebase](https://console.firebase.google.com/u/1/project/api-project-108380595987/testlab/histories/bh.edfd947f2636efe3), it looks like the problems occur due to slow load of Orders / Products lists either when performing search, or when just opening the screen.

This does not occur locally, so I'd blame the connectivity or lower Emulator performance on Firebase, which is not constant.

Instead of deleting the tests (which have a history of finding real issues, e.g. this crash: p1697012366493829-slack-C02H2SVA1L5), I'd suggest to account for the lower speed on CI, which I'm planning to do in [this branch](https://github.com/woocommerce/woocommerce-android/tree/ui-tests-add-conditional-waits) I just created.

In the worst case, we can disable these tests on CI, and still run them locally, since there's value in them.

WDYT?

**Comment by 0nko** [comment]
@pachlava Yeah, that sounds like a reasonable approach 👍. I only chose to remove them after some internal discussion when we noticed that it's becoming counterproductive. I'll close this PR in favour of your solution, thank you!
